### PR TITLE
feat: update pgwire and refactor pg auth handler

### PR DESCRIPTION
**Pull request by sunng87**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

- Updated postgres codec to deal with invalid input more aggressively, hoping to solve issues found in our cloud deployment cc @zyy17  https://github.com/sunng87/pgwire/commit/cbb767d9a9931352f413c27ca9c8609d6d1d76c6
- Refactoring postgresql auth handler to avoid clone of the whole metadata map

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Review by shuiyisong — APPROVED** [review]
LGTM

File: src/servers/Cargo.toml [hunk 1]
```diff
@@ -31,7 +31,7 @@ num_cpus = "1.13"
 once_cell = "1.16"
 openmetrics-parser = "0.4"
 opensrv-mysql = "0.3"
-pgwire = "0.5"
+pgwire = "0.6.1"
```
[the remark this answers is not in the document]
Comment (sunng87) [reply src/servers/Cargo.toml:34]: 0.6 has a minor issue so here I specify 0.6.1 to ensure required version is >= 0.6.1

File: src/servers/src/postgres/auth_handler.rs [hunk 2]
```diff
@@ -33,18 +33,41 @@ struct PgPwdVerifier {
     user_provider: Option<UserProviderRef>,
 }
 
+#[allow(dead_code)]
```
[the remark this answers is not in the document]
Comment (sunng87) [reply src/servers/src/postgres/auth_handler.rs:36]: `database` and `host` fields are not used for now, so I add this to suppress warnings.

**Review by Fengys123 — APPROVED** [review]
LGTM

**Pull request merged**